### ISSUE TITLE
ads-github-fetch-all-upstreams -vcu foo results in "line 1067: test: 7: unary operator expected" when github host is inaccessible

### ISSUE THREAD

**Issue by salewski** [salewski/ads-github-tools#34]
Looks like the program is attempting to use the total page count number without first verifying that it has been obtained. This was observed when working on a network that was dropping all traffic that was outbound to the Internet:

<pre>
    $ <strong>ads-github-fetch-all-upstreams -vcu</strong>
    ads-github-fetch-all-upstreams (info): verbose output requested; enabling info-level output
    ads-github-fetch-all-upstreams (info): configured GitHub API base URL: "https://api.github.com/"
    ads-github-fetch-all-upstreams (info): configured HTTP 'Accept:' header for GitHub API version: "Accept: application/vnd.github.v3+json"
    curl: (7) Failed to connect to api.github.com port 443: Connection timed out
    ads-github-fetch-all-upstreams (error): was unable to obtain pagination info for repos; bailing out
    ads-github-fetch-all-upstreams (info): requesting /user/repos page 1 of
    curl: (7) Failed to connect to api.github.com port 443: Connection timed out
    /path/to/bin/ads-github-fetch-all-upstreams: line 1067: test: 1: unary operator expected
    ads-github-fetch-all-upstreams (info): requesting /user/repos page 2 of
    curl: (7) Failed to connect to api.github.com port 443: Connection timed out
    /path/to/bin/ads-github-fetch-all-upstreams: line 1067: test: 2: unary operator expected
    ...
</pre>

<pre>
    $ <strong>ads-github-fetch-all-upstreams --version</strong>
    ads-github-fetch-all-upstreams 0.2.0  (built: 2017-02-26 16:52:07)
    Copyright (C) 2016, 2017 Alan D. Salewski
    License GPLv2+: GNU GPL version 2 or later <http://gnu.org/licenses/gpl.html>.
    This is free software: you are free to change and redistribute it.
    There is NO WARRANTY, to the extent permitted by law.

    Written by Alan D. Salewski.
</pre>

**Comment by salewski** [comment]
I just encountered what I think is a variation of this same issue in response to a 500 "Internal Server Error" from the upstream GitHub service:
```
ads-github-fetch-all-upstreams (info): invoking 'git fetch upstream' for repo "cl-graph"
ads-github-fetch-all-upstreams (info): no subdirectory exists for repo name "cl-hh-aws"; --clone-if-missing specified, so will clone repo from GitHub
git command: ['git', 'config', 'hub.username'] {'stderr': -1, 'stdout': -1}
git command: ['git', 'config', 'hub.oauthtoken'] {'stderr': -1, 'stdout': -1}
git command: ['git', 'config', 'hub.upstream'] {'stderr': -1, 'stdout': -1}
git command: ['git', 'config', 'hub.forkrepo'] {'stderr': -1, 'stdout': -1}
git command: ['git', 'config', 'hub.forkremote'] {'stderr': -1, 'stdout': -1}
git command: ['git', 'config', 'hub.pullbase'] {'stderr': -1, 'stdout': -1}
git command: ['git', 'config', 'hub.urltype'] {'stderr': -1, 'stdout': -1}
git command: ['git', 'config', 'hub.baseurl'] {'stderr': -1, 'stdout': -1}
git command: ['git', 'config', '--bool', 'hub.forcerebase'] {'stderr': -1, 'stdout': -1}
git command: ['git', 'config', '--bool', 'hub.triangular'] {'stderr': -1, 'stdout': -1}
git command: ['git', 'config', 'hub.password'] {'stderr': -1, 'stdout': -1}
Request: GET https://api.github.com/repos/salewski/cl-hh-aws?
None
https://api.github.com/repos/salewski/cl-hh-aws?
Content-length: 0
Content-type: application/json
Authorization: <hidden>
Accept: application/vnd.github.v3+json
None
Error: HTTP Error 500: Internal Server Error
Error: https://api.github.com/repos/salewski/cl-hh-aws?
Error: <httplib.HTTPMessage instance at 0x7f92ad34f200>
Error:
ads-github-fetch-all-upstreams (error): was error while attempting to clone repo "cl-hh-aws" from GitHub; bailing out
ads-github-merge-all-upstreams (info): verbose output requested; enabling info-level output
ads-github-merge-all-upstreams (info): "push" behavior requested; will push to 'origin' for clean projects that have changes merged during this run
ads-github-merge-all-upstreams (info): configured GitHub API base URL: "https://api.github.com/"
ads-github-merge-all-upstreams (info): configured HTTP 'Accept:' header for GitHub API version: "Accept: application/vnd.github.v3+json"
ads-github-merge-all-upstreams (error): program received a HTTP 5xx "Server Error" status codes; bailing out
    If you encounter this error, it suggests there is a problem with the upstream GitHub service;
    please wait a while and try your request later.
    Full HTTP response headers:
/path/to/ads-github-tools/bin/ads-github-merge-all-upstreams: line 696: printf: `        HTTP/1.1 500 Internal Server Error
        Server: GitHub.com
        Date: Mon, 16 Jul 2018 18:07:50 GMT
        Content-Type: application/octet-stream
        X-GitHub-Request-Id: xxxx:xxxx:xxxxxxx:xxxxxxxx:xxxxxxxx': not a valid identifier
ads-github-merge-all-upstreams (info): requesting /user/repos page 1 of
/path/to/ads-github-tools/bin/ads-github-merge-all-upstreams: line 1524: test: 1: unary operator expected
ads-github-merge-all-upstreams (info): requesting /user/repos page 2 of
/path/to/ads-github-tools/bin/ads-github-merge-all-upstreams: line 1524: test: 2: unary operator expected
ads-github-merge-all-upstreams (info): requesting /user/repos page 3 of
...
```

**Comment by salewski** [comment]
This issue as been fixed for 'ads-github-tools-0.3.1'

[The blurb about it](../blob/master/NEWS#issue-34-ads-github-fetch-all-upstreams--vcu-foo-results-in-line-1067-test-7-unary-operator-expected-when-github-host-is-inaccessible) from our ['NEWS'](../blob/master/NEWS) file:
```
*** issue 34: 'ads-github-fetch-all-upstreams -vcu foo results in "line 1067: test: 7: unary operator expected" when github host is inaccessible'

    https://github.com/salewski/ads-github-tools/issues/34

    This issue was partially fixed by the above changes for issue #33, but a
    check specific to this issue has been added to verify that the obtained
    page count value contains all digits. This keeps any emitted error
    messages as close as possible to the source of the problem.
```
